collective/auditlog/action.py

```diff
@@ -212,8 +209,12 @@ def getLogEntry(self):
                 return True
             obj = relationships[0]
 
-        data.update(self._getObjectInfo(obj))
+        data = self._getObjectInfo(obj)
```

Comment:
Working on dictionaries coming from another scope can be surprisingly dangerous:
```
>>> def a(d={}): return d
>>> a()["foo"] = "bar"
>>> a()
{'foo': 'bar'}
```
Is there a motivation behind this change?
If not I would keep the logic we had before or at least I would call `deepcopy`.

Reply:
I think the problem with your example is that the dictionary is created already on module import. You then modify then the module-level dict. ``getObjectInfo`` creates a new one every time it's called.
The motivation was mostly that previously the dict was created and filled with values for the keys ``action``, ``info`` and ``working_copy`` beforehand and then updated and eventually overwritten with the dict.update method.
It was actually not an issue because ``getObjectInfo`` doesn't use any of these keys. But I was playing with the idea to put the fallbacks for path (id) and type (dotted path of object) into the ``info`` key instead of path and id. (Ideally I would create new database columns, but I want to avoid to create a database schema upgrade step).

Reply:
To be on the safe side, a ``data = {}.update(self._getObjectInfo(obj))`` should do it? But I think it is not necessary.

Reply:
I was just pointing out that generally speaking the code as it was was safer. 
If you do:
```
data = {}.update(self._getObjectInfo(obj))
```
then `data` will be `None`.